jp-ag/jplox
# Refactor/build and structure

**Pull Request Description**

### Changes Made
- [x] **Maven Project Setup:** Configured the project structure using Maven conventions.
- [x] **JUnit Support:** Integrated JUnit for unit testing.

### Details
1. **Maven Structure:**
   - Moved the source code to `src/main/java` and the test code to `test/java`.
   - Added the `exec-maven-plugin` to facilitate running the Java application.

2. **JUnit Integration:**
   - Included the JUnit dependency in the `pom.xml` file.
   - Configured the `test/java` directory for test sources.
   - Developers can now write and run JUnit tests for the project.

### How to Test
1. Ensure Maven is installed (`mvn -v`).
2. Run `mvn test` to execute JUnit tests.
3. Run `mvn exec:java` to run the Java application.

### Additional Notes
- The main class for the application is `com.codinginterpreters.lox.Lox`.
- Check the Maven output for any compilation or execution errors.
